src/smartpeak/source/core/Utilities.cpp

```diff
@@ -500,78 +497,163 @@ namespace SmartPeak
     return false;
   }
 
-  std::array<std::vector<std::string>, 4> Utilities::getPathnameContent(
-    const std::string& pathname,
-    const bool asc
-  )
-  {
-    std::array<std::vector<std::string>, 4> content;
-    boost::system::error_code ec;
+std::array<std::vector<std::string>, 4> Utilities::getFolderContents(
+  const std::filesystem::path& folder_path,
+  const std::tuple<std::string, std::string>& sorting)
 
-    fs::directory_iterator it = fs::directory_iterator(fs::path(pathname), ec);
-    if (ec.value()) {
-      return content;
+  {
+     // name, ext, size, date
+    std::array<std::vector<std::string>, 4> directory_entries;
+    std::vector<std::tuple<std::string, uintmax_t, std::string, std::time_t>> entries_temp;
+    
+    for (auto & p : std::filesystem::directory_iterator(folder_path))
+    {
+      if (p.is_regular_file() && (p.path().filename().string().at(0) != '.') )
+      {
+        auto time           = std::filesystem::last_write_time(p.path());
+        auto sctp = std::chrono::time_point_cast<std::chrono::system_clock::duration>(time - std::filesystem::file_time_type::clock::now() + std::chrono::system_clock::now());
+        std::time_t cftime = std::chrono::system_clock::to_time_t(sctp);
+        //std::time_t cftime  = decltype(time)::clock::to_time_t(time);
+        entries_temp.push_back(std::make_tuple(p.path().filename(), p.file_size(), p.path().extension(), cftime));
+      }
+       else if (p.is_directory())
+       {
+         auto time          = std::filesystem::last_write_time(p.path());
+         auto sctp = std::chrono::time_point_cast<std::chrono::system_clock::duration>(time - std::filesystem::file_time_type::clock::now() + std::chrono::system_clock::now());
+         std::time_t cftime = std::chrono::system_clock::to_time_t(sctp);
+         //std::time_t cftime = decltype(time)::clock::to_time_t(time);
+         entries_temp.push_back(std::make_tuple(p.path().filename(), 0, "Directory", cftime));
+       }
     }
-    const fs::path p(pathname);
-    fs::directory_iterator it_end = fs::directory_iterator();
-
-    for ( ; it != it_end; it++) {
-      const fs::directory_entry& entry = *it;
-
-      if (!exists(entry.path())) { // protects from e.g. "dangling" symbolic links
-        LOGD << "Path does not exist. Skipping: " << entry.path();
-        continue;
+    
+    
+    if (!entries_temp.empty() || entries_temp.size() >= 2)
+    {
+      // sorting by filename
```

Comment:
Have you checked to see if sorting is possible using the the `tables` API of imgui?  If it is, this might not be needed to achieve this functionality as the FilePicker is currently using the legacy `columns` API and we would want to transition everything to the `tables` API and add in sorting, filtering, and searching support on the tables in the future.  Sorting, filtering, and searching are the most requested features for the GUI at the moment.

Reply:
It is possible with the current experimental `tables` API release. Would we want to have the transition seen in this PR or probably in a separate branch?

Reply:
Separate Branch.  See the new issue for this here: https://biosustain-dev.atlassian.net/browse/AUT-587.